### ISSUE TITLE
What should I do if I want to use reco loss on multiple GPUs.

### ISSUE THREAD

**Issue by lyccol** [lorenmt/reco#27]
What should I do if I want to use reco loss on multiple GPUs.

**Comment by lorenmt** [comment]
I would suggest check out accelerate framework: https://github.com/huggingface/accelerate

**Comment by lyccol** [comment]
Thanks for your reply, I reimplemented a multi-GPU code using DDP.

Another problem is that I found that randomcrop will crop out the valid pixels in partial_p0.

This will cause reco loss = 0 when I train partial_p0 data. At this point I need to set find_unused_parameters = True to continue training, but this greatly affects the training efficiency. Do you have any suitable suggestions?

**Comment by lorenmt** [comment]
I thought ReCo should also use highly confident pseudo-labels, and considering we are learning in a batch-wise manner, so it shouldn't be that common to have invalid loss everywhere?
